llama_index/retrievers/image_retriever.py

```diff
@@ -0,0 +1,75 @@
+"""Base vector store index query."""
+
+from abc import abstractmethod
+from typing import Any, Dict, List, Optional
+
+from llama_index.constants import DEFAULT_SIMILARITY_TOP_K
+from llama_index.indices.base_retriever import BaseRetriever
+from llama_index.indices.query.schema import QueryBundle
+from llama_index.schema import NodeWithScore
+from llama_index.vector_stores.types import (
+    MetadataFilters,
+)
+
+
+class BaseImageRetriever(BaseRetriever):
+    """Base Image retriever Abastraction.
+
+
+    Args:
+        BaseImageRetriever
+        similarity_top_k (int): number of top k results to return.
+        filters (Optional[MetadataFilters]): metadata filters, defaults to None
+    """
+
+    def __init__(
+        self,
+        image_similarity_top_k: int = DEFAULT_SIMILARITY_TOP_K,
+        filters: Optional[MetadataFilters] = None,
+        **kwargs: Any,
+    ) -> None:
+        """Initialize params."""
+        self._image_similarity_top_k = image_similarity_top_k
+        self._filters = filters
+        self._kwargs: Dict[str, Any] = kwargs
+
+    @property
+    def image_similarity_top_k(self) -> int:
+        """Return similarity top k."""
+        return self._image_similarity_top_k
+
+    @classmethod
+    def class_name(cls) -> str:
+        return "BaseImageRetriever"
+
+    @abstractmethod
+    def _retrieve(
+        self,
+        query_bundle: QueryBundle,
```

Comment:
we probably need to modify query bundle to include images

We could even override `retrieve()` instead of `_retrieve()` and do some checks for at least `query_str` or `image(s?)` being populated on the query bundle

Reply:
I would say let us do single image input retrieval at this stage.
if users want to do multiple images retrieval, they can call this multiple times

Reply:
i think i agree with @logan-markewich. we can modify str_or_query_bundle to be a composite type of Union[str, Image (?), QueryBundle] 

also in that case it's not clear we actually need to subclass BaseRetriever